# HG changeset patch
# User Mike Miller <mtmiller@ieee.org>
# Date 1370061020 14400
#      Sat Jun 01 00:30:20 2013 -0400
# Node ID 9ce08a1efc290ce75e04488de6d78f7e81d8f2a8
# Parent  6b00c6bff60c19aac33c90ddd0b3f4e4c0a315be
fix strsplit input validation

* strsplit.m: Fix input validation when called with no arguments.

diff --git a/scripts/strings/strsplit.m b/scripts/strings/strsplit.m
--- a/scripts/strings/strsplit.m
+++ b/scripts/strings/strsplit.m
@@ -140,17 +140,19 @@
 
 function [result, matches] = strsplit (str, del, varargin)
 
   args.collapsedelimiters = true;
   args.delimitertype = "default";
 
   [reg, params] = parseparams (varargin);
 
-  if (numel (reg) > 1)
+  if (nargin < 1)
+    print_usage ();
+  elseif (numel (reg) > 1)
     print_usage ();
   elseif (numel (reg) == 1)
     if (islogical (reg{1}) || isnumeric (reg{1}))
       args.collapsedelimiters = reg{1};
     else
       print_usage ();
     endif
   endif
@@ -191,19 +193,17 @@ function [result, matches] = strsplit (s
       del = {" ","\f","\n","\r","\t","\v"};
     elseif (strncmpi (args.delimitertype, "legacy", numel (args.delimitertype)))
       del = " \f\n\r\t\v";
     else
       del = "\\s";
     endif
   endif
 
-  if (nargin < 1)
-    print_usage ();
-  elseif (! ischar (str) || (! ischar (del) && ! iscellstr (del)))
+  if (! ischar (str) || (! ischar (del) && ! iscellstr (del)))
     error ("strsplit: S and DEL must be string values");
   elseif (! isscalar (args.collapsedelimiters))
     error ("strsplit: COLLAPSEDELIMITERS must be a scalar value");
   endif
 
   if (strncmpi (args.delimitertype, "simple", length_deltype))
     if (iscellstr (del))
       del = cellfun (@do_string_escapes, del, "uniformoutput", false);
